nautobot/core/views/__init__.py

```diff
@@ -144,6 +146,34 @@ def get(self, request, *args, **kwargs):
         return self.render_to_response(context)
 
 
+class AppDocsView(LoginRequiredMixin, View):
+    """
+    Serve documentation files for any pip-installed app from inside the package,
+    only for authenticated users.
+    """
+
+    def get(self, request, app_name, path="index.html"):
+        try:
+            base_dir = resources.files(app_name)
+        except ModuleNotFoundError:
+            return JsonResponse({"detail": f"App {app_name} not found."}, status=404)
+
+        # Dir to documentation inside the package
+        docs_dir = base_dir / "docs"
+
+        # Normalize path to avoid (../) etc.
```

Comment:
Can we add some security-related tests validating correct behavior on various attempted malicious inputs?